### PR TITLE
`General`: Document AI service pipelines

### PR DESCRIPTION
**Pull request by bassner**
## Summary

Publishes a code-grounded pipeline and lifecycle reference in each owning EduTelligence service and makes those references discoverable from the suite README.

## Changes

- **Iris/Pyris:** documents pipeline hierarchy, route-specific execution, callbacks, chat modes, ingestion and deletion jobs, RAG retrieval, and Iris-owned Memiris integration.
- **Athena:** adds the end-to-end assessment-module request lifecycle, including manager forwarding, decorators, AI-selection decisions, and the authenticated Artemis repository boundary.
- **AtlasML:** replaces illustrative pipeline material with current workflow, clustering, mapping, relationship, feedback, model, dimension, and device behavior.
- **Logos:** documents classification and direct-model modes across classification, scheduling, context resolution, forwarding, and completion.
- **Memiris:** documents the reusable creation/sleep library separately from Iris runtime scheduling, canonicalizes the README filename, and keeps the Poetry package metadata aligned with that path.
- Adds canonical pipeline and lifecycle links to the root and Logos READMEs.

## Verification

- Iris docs: `npm run typecheck`, `npm run build`
- Athena docs: `npm run typecheck`, `npm run build`
- Atlas docs: `npm run typecheck`, `npm run build`
- Memiris: `poetry check --lock`, `poetry build`, package installation on Python 3.13, and all 68 tests
- Iris: package installation with the local Memiris dependency on Python 3.13 and all 268 tests
- Generated routes verified for the Iris pipeline and RAG pages, Athena lifecycle page, and AtlasML pipeline page.
- Root pre-commit hooks passed for all changed files, including Iris, Athena, Memiris, and Logos sub-project hooks.
- `git diff --check` passed.
- Existing published Iris and AtlasML links were checked successfully; the new Athena lifecycle route was verified from the generated build output.

## Summary by CodeRabbit

* **Documentation**
  * Added a centralized “Pipeline and lifecycle references” index.
  * Expanded Athena request-lifecycle documentation and linked it in the dev sidebar.
  * Rewrote Iris pipeline and RAG workflow docs for clearer execution/lifecycle guidance.
  * Reworked Atlas ML pipelines into an implementation-focused code reference.
  * Updated Logos request-pipeline and scheduler documentation.
  * Refreshed Memiris documentation with new reusable-library usage and architecture details; removed outdated Memiris README content.
* **Bug Fixes**
  * Corrected README file inclusion for Docker builds and package metadata by fixing README casing/reference.

### CODE REVIEW

File: logos/logos-orchestrator/src/logos/pipeline/README.md [hunk 1]
```diff
@@ -1,202 +1,44 @@
-# Request Pipeline
-
-## Overview
-
-The Request Pipeline orchestrates the lifecycle of a request from entry to execution. It decouples the three main stages of request handling:
```
Comment (wasnertobias) [inline logos/logos-orchestrator/src/logos/pipeline/README.md:5]: I would like to leave this documentation

Comment (bassner) [reply logos/logos-orchestrator/src/logos/pipeline/README.md:5]: Addressed in 223a523fd. I kept the full detailed pipeline README and updated its stale scheduler/runtime claims in place (active ClassificationCorrectingScheduler, ETTFT ablation, strict queue priorities, current facades, and the ContextResolver/Executor ownership split) instead of replacing the documentation.

**Comment by bassner** [comment]
@Claudia-Anthropica review

@coderabbitai review

**Comment by Claudia-Anthropica** [comment]
@bassner sure thing, diving in now :rocket:

**Comment by bassner** [comment]
@Claudia-Anthropica review

@coderabbitai review

**Comment by Claudia-Anthropica** [comment]
@bassner sure thing, diving in now :rocket:

File: athena/docs/docs/dev/module/request-lifecycle.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+- `X-Module-Config` for configuration evaluated by the module's schema provider;
+- `X-Experiment-ID`, `X-Module-Configuration-ID`, and `X-Run-ID` for experiment context; and
+- `X-Server-URL`, the Artemis/LMS base URL used by a module that needs to call back into the LMS.
+
+The manager authenticates the module-to-module call separately. For programming modules it also adds the repository-authorization secret associated with the supplied LMS URL; this value is not returned to Artemis or documented as a client configuration value.
+
+## 2. The module contract processes the request
+
+The Athena package exposes FastAPI endpoints through decorators in `athena/athena/endpoints.py`. The decorators authenticate the forwarded request, merge stored metadata, persist exercises/submissions/feedback where appropriate, deserialize module configuration, and call the implementation.
```
Comment (Claudia-Anthropica) [inline athena/docs/docs/dev/module/request-lifecycle.md]: @bassner [medium] This overstates the decorator authentication contract. The operational decorators use `@authenticated`, but `@config_schema_provider` registers `GET /config_schema` without it, even though that decorator is included in the table below; narrow this statement or document the exception explicitly.

<details><summary>🤖 Prompt for AI agents</summary>

In `athena/docs/docs/dev/module/request-lifecycle.md`, the module-contract section incorrectly says all endpoint decorators authenticate forwarded requests even though `config_schema_provider` does not. Rewrite the authentication statement and configuration row to distinguish the unauthenticated schema endpoint from the authenticated operational decorators.

</details>

File: logos/logos-orchestrator/src/logos/pipeline/README.md [hunk 3]
```diff
@@ -45,17 +45,17 @@ The Request Pipeline orchestrates the lifecycle of a request from entry to execu
 ║  ┌──────────────────┐   ┌───────────────────────┐   ┌────────────────── ┐ ║
 ║  │ executor.py      │   │  Scheduler Layer      │   │context_resolver.py║ ║
 ║  │                  │   │                       │   │                   │ ║
-║  │ • Model lookup   │   │scheduler_interface.py │   │ • Provider        │ ║
-║  │ • Provider info  │   │  SchedulerInterface   │   │   resolution      │ ║
-║  │ • API keys       │   │         │             │   │ • Endpoint        │ ║
-║  │ • Backend calls  │   │         ▼             │   │   lookup          │ ║
-║  │ • Streaming      │   │  base_scheduler.py    │   │                   │ ║
+║  │ • HTTP requests  │   │scheduler_interface.py │   │ • Model/provider  │ ║
```
Comment (Claudia-Anthropica) [inline logos/logos-orchestrator/src/logos/pipeline/README.md]: @bassner [medium] The ownership split is still contradictory here and in the `pipeline.py` component description below. `RequestPipeline.process()` stops after classification, scheduling, and context resolution; `main.py` then calls `_sync_response` or `_streaming_response`, which performs the upstream execution, so the diagram must not show `RequestPipeline` invoking `Executor`.

<details><summary>🤖 Prompt for AI agents</summary>

In `logos/logos-orchestrator/src/logos/pipeline/README.md`, the architecture diagram and `pipeline.py` component section incorrectly say `RequestPipeline` executes the upstream request. Update both sections to show that `RequestPipeline.process()` returns the resolved execution context and `main.py` owns the subsequent sync or streaming execution through `Executor`.

</details>

File: atlas/docs/docs/dev/code-reference/ml-pipelines.md [hunk 4]
```diff
@@ -1,721 +1,54 @@
...
+The workflow contains an implemented centroid-update path; it is not a generic learning or caching proposal.
 
-    # 2. Move query closer to accepted
-    adjusted_embedding = []
-    learning_rate = 0.1
-    for q, a in zip(query_embedding, accepted_embedding):
-        adjusted = q + learning_rate * (a - q)
-        adjusted_embedding.append(adjusted)
+- `instructor_feedback_on_new_text` runs after a new exercise with assigned competencies is stored. It adds the exercise embedding to each relevant cluster centroid.
+- `instructor_feedback` runs when an existing exercise is updated. It compares the stored competency IDs with the updated IDs. For both removed and newly assigned competencies, it passes the newly generated updated exercise embedding to the centroid helper; it does not retrieve or use the exercise's prior stored vector for removals.
```
Comment (Claudia-Anthropica) [inline atlas/docs/docs/dev/code-reference/ml-pipelines.md:45]: @bassner [low] This misses the empty-update path: `instructor_feedback()` returns immediately when `updated_exercise.competencies` is empty, so removing the final or all assignments never reaches the centroid-removal helper. Document that current limitation so this code-grounded description does not imply every removal is processed.

<details><summary>🤖 Prompt for AI agents</summary>

In `atlas/docs/docs/dev/code-reference/ml-pipelines.md`, the `instructor_feedback` description omits that an update with an empty competency list returns before processing removals. Add that exception to the centroid-update documentation without implying that the last assignment updates its old centroid.

</details>

**Review by Claudia-Anthropica — CHANGES_REQUESTED** [review]
@bassner The refreshed references are mostly well grounded, but Athena overstates endpoint authentication and Logos still contradicts the actual ContextResolver/Executor ownership split. Please address the inline notes.

**Comment by bassner** [comment]
Pushed `bee54bf5e`: corrected Athena's unauthenticated schema-endpoint exception, Logos's `main.py`/Executor ownership, and Atlas's empty-competency centroid limitation. Pre-commit passed for Athena and Logos; both affected docs typechecks pass. The remaining worker-image failure is Harbor's 100 GiB quota, not this documentation diff.

@claudia review
@coderabbitai review

**Comment by bassner** [comment]
@claudia review\n@coderabbitai review\n\nLatest follow-up corrects the two remaining documentation claims: source data are scalar identifiers/page metadata while Weaviate cross-collection references are disabled, and the current default branch does not yet guarantee composition of every supplied retrieval scope. Pre-commit passes.

**Review by Claudia-Anthropica — CHANGES_REQUESTED** [review]
@bassner The Athena and AtlasML follow-ups are fixed. The Logos ownership prose is corrected, but its architecture diagram still shows `RequestPipeline` invoking `Executor`; I’ve left the remaining detail in the existing thread.

File: athena/docs/docs/dev/module/request-lifecycle.md (hunk 2 above)
Comment (Claudia-Anthropica) [reply athena/docs/docs/dev/module/request-lifecycle.md]: @bassner Thanks, the operational/config-schema authentication distinction is explicit now. This is fixed.

File: logos/logos-orchestrator/src/logos/pipeline/README.md (hunk 3 above)
Comment (Claudia-Anthropica) [reply logos/logos-orchestrator/src/logos/pipeline/README.md]: @bassner The prose is fixed, but the architecture diagram still puts `executor.py` under arrows from `pipeline.py - RequestPipeline` (lines 43–46), so it continues to show the wrong ownership. Please move `Executor` to the HTTP-layer execution path after `RequestPipeline.process()` returns.

File: atlas/docs/docs/dev/code-reference/ml-pipelines.md (hunk 4 above)
Comment (Claudia-Anthropica) [reply atlas/docs/docs/dev/code-reference/ml-pipelines.md:45]: @bassner Thanks, the empty-assignment limitation is documented now. This is fixed.

**Comment by bassner** [comment]
@claudia review

The remaining Logos ownership diagram now shows `RequestPipeline.process()` returning a `RequestContext` to `main.py`, with `executor.py` owned by the HTTP-layer execution path.

**Comment by bassner** [comment]
@claudia review
@coderabbitai review

Exact head is now `0bb063c523`. The final diagram ownership fix has also passed an independent exact-head review: `RequestPipeline.process()` returns `PipelineResult`, and only `main.py` subsequently enters the separate executor path. No actionable findings remain.

**Review by Claudia-Anthropica — DISMISSED** [review]
@bassner The requested ownership and lifecycle clarifications are resolved now. The remaining CI failures are unrelated Harbor capacity errors, so this is good to merge.

**Comment by bassner** [comment]
Corrected the Logos documentation against the actual implementation at `64a3e6516a`: registry RPC vs executor responsibilities, `PipelineResult`/`execution_context`, context resolution, response preparation/persistence, model-only queueing, catch-all routes, release semantics, and the broken SDI reference. Logos pre-commit, relative-link checks, repository CI, an independent exact-head review, and rendered GitHub visual verification all passed.

@claudia review
@coderabbitai review

**Review by Claudia-Anthropica — APPROVED** [review]
@bassner The corrected Logos lifecycle reference now matches the implementation boundaries, and the broader service documentation remains well grounded. Looks good to merge.
